### PR TITLE
feat(schematics): Add ability to ng update from Nx 1.x

### LINKED ISSUE #476: Can't run update on CLI v6 app
Hi

With angular cli V6 app and nx 6.0.0-alpha.1 I'm getting this error when running `npm run update:check`

**Error: ENOENT: no such file or directory, open '.angular-cli.json'**

This also happens in other commands (e.g. postinstall)

I thinks it's related to this https://github.com/nrwl/nx/blob/master/packages/schematics/src/utils/fileutils.ts#L55

### PR DESCRIPTION
**Pull request by FrozenPandaz**
This adds the new Angular CLI Migrations to Nx!

This allows a user to upgrade from a Nx 1.x workspace to a Nx 6.0 Workspace!

Ng Update Specifications: https://github.com/angular/devkit/blob/master/docs/specifications/update.md

### Update Procedure:
1. Install the latest version of `@nrwl/schematics`
```
yarn add @nrwl/schematics@latest
```
* On Local this will be 
```
yarn add ../../nx/builds/packages/schematics
```

2. Run the last Nx update
```
yarn update
```
3. Update `@angular/cli`
```
yarn add @angular/cli@latest -D
ng update @angular/cli
```

4. Update `@nrwl/schematics`
```
ng update @nrwl/schematics
```
* On Local this will be
```
ng @schematics/update:migrate --package @nrwl/schematics --collection @nrwl/schematics/migrations/migrations.json --from 1.0.0 --to 6.0.0
```

After all this, the project should be mostly functioning.

Note:
It might be ugly.

Fixes #476

### CODE REVIEW

File: packages/schematics/src/command-line/nx.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import * as yargs from 'yargs';
 import { affected } from './affected';
 import { format } from './format';
 import { update } from './update';
-import { patchNg } from './patch-ng';
+// import { patchNg } from './patch-ng';
```
Comment (jschwarty) [inline packages/schematics/src/command-line/nx.ts]: Should this line just get removed since the `patch-ng` file is deleted?

**Comment by vsavkin** [comment]
I think we need to create a migration performing all the steps you outlined. So when I run: `yarn add @nrwl/schematics@latest AND yarn update`, it does all of them:

1. creates nx.json
2. installs @angular/cli@6.0.1
3. runs ng update
4. runs update @nrwl/schematics

And it should always reference specific version (e.g., 6.0.1, and not "latest").

This migration isn't moving Nx to the latest version of the CLI, but to CLI 6.0.1. All other updates of the CLI or Nx should be handled elsewhere.

File: packages/schematics/migrations/migrations.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "schematics": {
+    "migration-01": {
```
Comment (vsavkin) [inline packages/schematics/migrations/migrations.json]: maybe a better name?

File: packages/schematics/migrations/legacy-migrations/20180507-create-nx-json.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    writeFileSync(
+      'nx.json',
+      serializeJson({
+        npmScope: angularJson.project.npmScope,
+        projects: projects
+      })
+    );
+
+    updateJsonFile('package.json', json => {
+      json.scripts.postinstall = json.scripts.postinstall || 'echo 1';
```
Comment (vsavkin) [inline packages/schematics/migrations/legacy-migrations/20180507-create-nx-json.ts]: I deleted the postinstall hook. So you can set it to undefined.

File: packages/schematics/migrations/update-6-0-0/update-6-0-0.ts [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+  findFunctionCalls,
+  findSpecDeclaration,
+  findTsNodeRegisterExpression
+} from './helpers';
+
+import * as ts from 'typescript';
+import * as path from 'path';
+import * as fs from 'fs';
+
+function updateKarmaConfigs(host: Tree, context: SchematicContext) {
```
Comment (vsavkin) [inline packages/schematics/migrations/update-6-0-0/update-6-0-0.ts]: I don't think we can reuse the karma.conf.js. I would just replace it with the default karma.conf.js CLI generates for apps and libs

**Review by vsavkin — CHANGES_REQUESTED** [review]

**Comment by vsavkin** [comment]
Make sure the following is true:

```
create-nx-workspace proj // created using nx 1.0.3
ng add myapp --routing --directory=dir
ng add mylib --routing --lazy --parentModule=apps/dir/myapp/src/app/app.module.ts --directory=dir
yarn add @nrwl/schematics@next // or install using local package
yarn install
yarn update
```

and

```
ng new proj --collection=@nrwl/schematics // using schematics 6.0.0-alpha * and cli 6.0.1
ng add myapp --routing --directory=dir
ng add mylib --routing --lazy --parentModule=apps/dir/myapp/src/app/app.module.ts --directory=dir
```

Create same projects, in particular:

1. all tslint.json, package.json, and tsconfig.*.json, angular.json are the same (unless there is a good reason for them not to be)
2. all app.module.ts are the same

One of the limitations of CLI 6 is that project names cannot contain "/". This means that when updating, your should replace / with - in project names.

**Comment by juristr** [comment]
I need to update my nx workspace to ng6 as soon as possible because I need Angular elements in a project 😄.
Thus I started testing the upgrade, first by generating a new nx workspace using NX v1.

Now  I'm following the above steps to see whether I succeed upgrading the simple workspace for then replicating on my more complex setup.

Currently I'm stuck on the `yarn add @angular/cli@latest -D` step:
```
$ ./node_modules/.bin/nx postinstall
internal/modules/cjs/loader.js:550
    throw err;
    ^

Error: Cannot find module 'fs-extra'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:548:15)
```

**Comment by FrozenPandaz** [comment]
@juristr Sorry, that would be the instructions in the future once it is final. For now, you are correct, you would have to `yarn add @nrwl/schematics@next`.

**Comment by vsavkin** [comment]
merged here https://github.com/nrwl/nx/commit/8de81d6c16935393239b2d7093bc8e72ab251311

**Pull request closed without merging**